app/ModelFunctions/MOVFormat.php

```diff
@@ -2,38 +2,62 @@
 
 namespace App\ModelFunctions;
 
-use FFMpeg;
+use App\Exceptions\Internal\LycheeAssertionError;
+use FFMpeg\Exception\InvalidArgumentException;
+use FFMpeg\Format\Video\DefaultVideo;
 
-// Class for FFMpeg to convert files to mov format
-class MOVFormat extends FFMpeg\Format\Video\DefaultVideo
+/**
+ * Format class for FFMpeg to store a single video stream in a Quicktime (MOV) container.
+ */
+class MOVFormat extends DefaultVideo
 {
-	/**
-	 * @throws FFMpeg\Exception\InvalidArgumentException
-	 */
-	public function __construct(string $audioCodec = 'copy', string $videoCodec = 'copy')
+	public const FFMPEG_VIDEO_CODEC_ID = 'copy';
+	public const FFMPEG_AUDIO_CODEC_ID = 'copy';
+	public const FFMPEG_CONTAINER_ID = 'mov';
+
+	public function __construct()
 	{
-		$this
-			->setAudioCodec($audioCodec)
-			->setVideoCodec($videoCodec);
+		try {
+			$this
+				->setAudioCodec(self::FFMPEG_AUDIO_CODEC_ID)
+				->setVideoCodec(self::FFMPEG_VIDEO_CODEC_ID);
+		} catch (InvalidArgumentException $e) {
+			throw LycheeAssertionError::createFromUnexpectedException($e);
+		}
 	}
 
 	public function supportBFrames(): bool
 	{
 		return false;
 	}
 
+	/**
+	 * Returns the extra parameters to be added to the FFmpeg command line.
+	 *
+	 * Here we force FFmpeg to use the Quicktime Container format for output.
+	 *
+	 * TODO: Why do we enforce Apple Quicktime as the output container? This is suspicious.
+	 *
+	 * Note, that we only use this format to store extracted video streams
+	 * from Google Motions Pictures.
+	 * A Google Motion Picture uses AVC (H.264) as the codec in an MP4
+	 * (MPEG-4 Part 1) container.
+	 * We do want to re-pack it in another container?
```

Comment:
Yeah, I've wondered the same thing -- why are we not simply keeping it in mp4? Browsers seem to support mp4 just fine so that can't be it...

Reply:
I have no idea. If you trust MDN, then according to

 - https://developer.mozilla.org/en-US/docs/Web/Media/Formats/Containers#index_of_media_container_formats_file_types
 - https://developer.mozilla.org/en-US/docs/Web/Media/Formats/Video_codecs#common_codecs

AVC in an MP4 container is just fine, is standardized for the HTML5 `<video>`-tag and has even wider browser support. Maybe this dates back to Lychee v3 when Apple Quicktime was state of the art, because Safari took a leading role back then?

But I also believe to remember that our current web frontend does not use the HTML5 `<video>` tag but some external JS library to provide a video player in the browser. Maybe that JS library requires Quicktime? But then we should switch to a more modern HTML5 approach.

Reply:
I'm pretty sure we didn't support live photos in v3... Also, you are misremembering: our web frontend _does_ use the `<video>` tag and has no custom JS for playing videos.

The original PR is https://github.com/LycheeOrg/Lychee/pull/404, based on a preceding discussion in https://github.com/LycheeOrg/Lychee/issues/378. There is a comment there (https://github.com/LycheeOrg/Lychee/issues/378#issuecomment-548687276): _Package for viewing only supports mov files, hence the current version converts the mp4 file from a Google Motion Photo to mov while keeping codecs, resolution etc. (should be very fast)_. So I guess that explains it -- it's not the web browser, but the package for handling live photos that needs a MOV. Perhaps it got updated since 2019 to support MP4s as well but I can't be bothered to check (honestly, I don't care about live photos).

Reply:
I guess when the comment says "the package" it refers to this NPM package: `livephotoskit/livephotoskit`. Yes, I was wrong about the `<video>` tag, but this is what I meant when I said maybe some JS library requires Quicktime. We still use some 3rd party JS code to show the video player.

> honestly, I don't care about live photos

Me neither. Somehow I assume that the whole JS package might have become superfluous, but I don't know what it does specifically, why it has been in the first place, etc.  I won't change anything here. I only add a comment to the PHP class to explain some background.

Reply:
Fixed by https://github.com/LycheeOrg/Lychee/pull/1351/commits/efbe10b03c4a2b225e86397eff95ea3c5d7bf75f.